hpx/util/annotated_function.hpp

```diff
@@ -137,7 +137,6 @@ namespace hpx { namespace util
                 typename util::decay_unwrap<F>::type, Ts...>::type
             operator()(Ts && ... ts)
             {
-                annotate_function func(name_);
```

Comment:
This call is the whole point of having the `annotated_function` type. Removing this call renders it useless.

Reply:
@hkaiser What if we put `#if !defined(HPX_HAVE_APEX)` / `#endif` around it?  There is other code in HPX that will do the annotation correctly when the function happens to be of the type `annotated_function`.

Reply:
@hkaiser I'm not sure I understand. The `annotated_function` wrapper sets the annotation name in the creation of the task and does the 'right' thing. The call to annotate_function inside the `annotated_function` is doing what?

Reply:
@biddisco right, I misunderstood - I think all is well now.